### PR TITLE
fix(inputs.dns_query): Fill out additional record fields

### LINKED ISSUE #14956: Telegraf inputs.dns_query plugin does not return any data for SOA type RRs
### Relevant telegraf.conf

```toml
[[outputs.prometheus_client]]
  listen = ":4444"
  path = "/metrics"
  collectors_exclude = ["gocollector", "process"]
  export_timestamp = false

[agent]
  interval = "10s"
  round_interval = true
  metric_batch_size = 1000
  metric_buffer_limit = 10000
  collection_jitter = "0s"
  flush_interval = "5s"
  flush_jitter = "0s"
  precision = ""
  debug = false
  quiet = false
  logfile = "/tmp/aaa.log"
  hostname = ""
  omit_hostname = false

# SOA
[[inputs.dns_query]]
  servers = ["1.1.1.1"]
  network = "udp"
  timeout = 1
  port = 53
  record_type = "SOA"
  domains = [
              "example.com",
            ]
```

### Logs from Telegraf

```text
root@node_exp:~# tail -f  /tmp/aaa.log 

2024-03-08T17:17:53Z I! Starting Telegraf 1.29.5 brought to you by InfluxData the makers of InfluxDB
2024-03-08T17:17:53Z I! Available plugins: 241 inputs, 9 aggregators, 30 processors, 24 parsers, 60 outputs, 6 secret-stores
2024-03-08T17:17:53Z I! Loaded inputs: dns_query
2024-03-08T17:17:53Z I! Loaded aggregators: 
2024-03-08T17:17:53Z I! Loaded processors: 
2024-03-08T17:17:53Z I! Loaded secretstores: 
2024-03-08T17:17:53Z I! Loaded outputs: prometheus_client
2024-03-08T17:17:53Z I! Tags enabled: host=node_exp
2024-03-08T17:17:53Z I! [agent] Config: Interval:10s, Quiet:false, Hostname:"node_exp", Flush Interval:5s
2024-03-08T17:17:53Z D! [agent] Initializing plugins
2024-03-08T17:17:53Z D! [agent] Connecting outputs
2024-03-08T17:17:53Z D! [agent] Attempting connection to [outputs.prometheus_client]
2024-03-08T17:17:53Z I! [outputs.prometheus_client] Listening on http://[::]:4444/metrics
2024-03-08T17:17:53Z D! [agent] Successfully connected to outputs.prometheus_client
2024-03-08T17:17:53Z D! [agent] Starting service inputs
2024-03-08T17:17:58Z D! [outputs.prometheus_client] Buffer fullness: 0 / 10000 metrics
2024-03-08T17:18:03Z D! [outputs.prometheus_client] Wrote batch of 1 metrics in 307.246µs
2024-03-08T17:18:03Z D! [outputs.prometheus_client] Buffer fullness: 0 / 10000 metrics
2024-03-08T17:18:08Z D! [outputs.prometheus_client] Buffer fullness: 0 / 10000 metrics
2024-03-08T17:18:13Z D! [outputs.prometheus_client] Wrote batch of 1 metrics in 333.081µs
2024-03-08T17:18:13Z D! [outputs.prometheus_client] Buffer fullness: 0 / 10000 metrics
2024-03-08T17:18:18Z D! [outputs.prometheus_client] Buffer fullness: 0 / 10000 metrics


root@node_exp:~# curl localhost:4444/metrics
# HELP dns_query_rcode_value Telegraf collected metric
# TYPE dns_query_rcode_value untyped
dns_query_rcode_value{domain="example.com",host="node_exp",rcode="NOERROR",record_type="SOA",result="success",server="1.1.1.1"} 0
# HELP dns_query_result_code Telegraf collected metric
# TYPE dns_query_result_code untyped
dns_query_result_code{domain="example.com",host="node_exp",rcode="NOERROR",record_type="SOA",result="success",server="1.1.1.1"} 0
```

### System info

Telegraf 1.29.5

### Docker

_No response_

### Steps to reproduce

Copy the config file and run it with Telegraf 1.29.5

### Expected behavior

$ dig +short +multi @1.1.1.1 example.com soa
ns.icann.org. noc.dns.icann.org. 2024013026 7200 3600 1209600 3600

### Actual behavior

dns_query_rcode_value and dns_query_result_code retun 0

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by powersj**
## Checklist

- [x] No AI generated code was used in this PR

## Related issues

resolves #14956

### CODE REVIEW

File: plugins/inputs/dns_query/dns_query.go [hunk 1]
```diff
@@ -155,6 +155,24 @@ func (d *DNSQuery) query(domain string, server string) (map[string]interface{},
...
+		case "SOA":
+			if soa, ok := record.(*dns.SOA); ok {
+				fields["serial"] = soa.Serial
+				fields["refresh"] = soa.Refresh
+				fields["retry"] = soa.Retry
+				fields["expire"] = soa.Expire
+				fields["minttl"] = soa.Minttl
+			}
+		}
+	}
```
Comment (srebhan) [inline plugins/inputs/dns_query/dns_query.go:174]: How about
```suggestion
		switch x := record.(type) {
		case *dns.MX:
			fields["preference"] = x.Preference
		case *dns.SOA:
			fields["serial"] = x.Serial
			fields["refresh"] = x.Refresh
			fields["retry"] = x.Retry
			fields["expire"] = x.Expire
			fields["minttl"] = x.Minttl
		}
```

**Review by srebhan — COMMENTED** [review]
@powersj I do have one suggestion and wonder if it makes sense to also add the additional fields of all other supported type (see `parseRecordType()`)!?

**Comment by powersj** [comment]
> @powersj I do have one suggestion and wonder if it makes sense to also add the additional fields of all other supported type (see `parseRecordType()`)!?

For the majority of the other types there is nothing else or there is a `cdomain-name`, which I have assumed was the domain we are looking up and captured as a tag already. Is that your assessment as well?

**Comment by powersj** [comment]
@srebhan we don't do reverse looksups (e.g. ip -> dns name) so I believe this is ready to go.

**Review by srebhan — APPROVED** [review]
Thanks @powersj!

**Review by DStrand1 — APPROVED** [review]

**Pull request merged**